src/Interpreters/Context.h

```diff
@@ -447,7 +454,7 @@ class Context: public std::enable_shared_from_this<Context>
 
     void addWarningMessage(const String & msg) const;
 
-    VolumePtr setTemporaryStorage(const String & path, const String & policy_name = "");
+    VolumePtr setTemporaryStorage(const String & path, const String & policy_name = "", size_t max_size = 0);
```

Comment:
as far as I see this method is used only from server.cpp. may be make this argument not optional then? as there is not case when it is omitted.